wsi/drm/drmdisplaymanager.cpp

```diff
@@ -442,17 +446,73 @@ void DrmDisplayManager::IgnoreUpdates() {
   }
 }
 
-void DrmDisplayManager::setDrmMaster() {
-  int ret = drmSetMaster(fd_);
-  while (ret) {
-    usleep(10000);
-    ret = drmSetMaster(fd_);
+DrmDisplayManager::DrmMasterThread::DrmMasterThread(
+    DrmDisplayManager &drmdisplaymanager)
+    : HWCThread(-8, "DrmMasterThread"), manager(drmdisplaymanager) {
+  CTRACE();
+}
+
+DrmDisplayManager::DrmMasterThread::~DrmMasterThread() {
+  CTRACE();
+  HWCThread::Exit();
+}
+
+bool DrmDisplayManager::DrmMasterThread::StartSetting() {
+  if (!running) {
+    running = true;
+    return InitWorker();
+  } else {
+    Resume();
+    return true;
+  }
+}
+
+void DrmDisplayManager::DrmMasterThread::HandleWait() {
+  if (!need_drm_master_)
+    HWCThread::HandleWait();
+  usleep(10000);
+}
+
+void DrmDisplayManager::DrmMasterThread::HandleRoutine() {
+  if (need_drm_master_) {
+    int ret = drmSetMaster(manager.fd_);
     if (ret) {
       ETRACE("Failed to call drmSetMaster : %s", PRINTERROR());
+      need_drm_master_ = true;
+      manager.drm_master_ = false;
+    } else {
+      need_drm_master_ = false;
+      manager.drm_master_ = true;
     }
   }
 }
 
+void DrmDisplayManager::DrmMasterThread::NeedDrmMaster(bool need) {
+  need_drm_master_ = need;
+}
+
+void DrmDisplayManager::setDrmMaster() {
+  int ret = drmSetMaster(fd_);
+  if (ret) {
+    ETRACE("Failed to call drmSetMaster : %s", PRINTERROR());
+    drm_master_thread_->NeedDrmMaster(true);
```

Comment:
Do we need a new drm_master_thread_?

Reply:
The logic here does not make sure setDrmMaster is successful
However, we need to make sure it is successful
Suggest to keep origin logic instead of using a thread

Reply:
The origin logic will block to setDrmMaster and enter a endless loop, once kmscube/earlyevs still hold drmMaster and invoke HWC to setDrmMaster. Need a async thread to setDrmMaster. 
Once the variable need_drm_master still true, the thread will keep running for setDrmMaster